### PR TITLE
Inconsistent formatting of dates (EXPOSUREAPP-9127)

### LINKED ISSUE #3963: Inconsistent formatting of dates in app
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [ ] Bug is not already reported in another issue
See also https://github.com/corona-warn-app/cwa-app-android/issues/3830

## Technical details

- Android version: 11
- App version: 2.9.0

## Describe the bug

The format used to display dates in CWA in numerical format is inconsistent. In some places the user's preference which has been set via the devices locale settings is used, in other places the format is hard-coded to dd.mm.yy or dd.mm.yyyy, with 2-digit and 4-digit year formats being used on the same screen.

With the locale set to English (United States) the following different formats are found:

Section | Format | Example
--- | --- | ---
Certificates - date of birth | dd.mm.yyyy | PersonDetailsFragment_cwa_2.png
Certificates - vaccination | dd.mm.yy | PersonDetailsFragment_cwa_2.png
Certificates - test | dd.mm.yyyy | PersonDetailsFragment_cwa_2.png
Rapid Test Profile - date of birth | dd.mm.yyyy | RATProfileQrCodeFragment.png
PCR Test | dd.mm.yy | HomeFragment_test_positive.png
Contact Journal | mm/dd/yy | ContactDiaryDayFragment_locations_data.png

PersonDetailsFragment_cwa_2

RATProfileQrCodeFragment

HomeFragment_test_positive

ContactDiaryDayFragment_locations_no_data

## Steps to reproduce the issue

1. Settings > System > Languages and input > Languages<br>set to English (United States)
2. Settings > System > Date & time<br>select "Use locale default"/ disable "Use 24-hour format"
3. Connect mobile device to Android Studio
4. In Android Studio Build > Select BuildVariant deviceForTestersDebug
5. run src/androidTest/java/de/rki/coronawarnapp on emulator
6. View > Tools Windows > Device File Explorer
7. Go to /data/data/de.rki.coronawarnapp.test/screenshots
8. View screenshots and note discrepancies in the way that the date is shown

## Expected behavior

The display of dates in numerical format in CWA should be consistent and preferably the locale format should be used.

## Possible Fix

Standardize date output across the app.

---
Internal Tracking ID: [EXPOSUREAPP-9127](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9127) Android
Internal Tracking ID: [EXPOSUREAPP-9220](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9220) Design
Internal Tracking ID: [EXPOSUREAPP-9238](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9238) iOS

### PR DESCRIPTION
**Pull request by marcauberer**
Addresses [EXPOSUREAPP-9127](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9127).

- Replaces the static date formats `DD.MM.YY` and `DD.MM.YYYY` with pre-defined short and medium localized date formats
- Unifies date formatting across the app
- Aligns date formatting with iOS

### CODE REVIEW

**Review by MikeMcC399 — DISMISSED** [review]
I checked again, and it looks good to go!

- Screenshots ✅ 
- Rapid Text Profile ✅ 
- COVID Test Certificate ✅ 

Thank you!

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/CwaUserCard.kt [hunk 1]
```diff
@@ -41,9 +43,9 @@ class CwaUserCard(parent: ViewGroup) :
 
     private fun formatBirthDate(dateOfBirthFormatted: String): String =
         try {
-            dateOfBirthFormatted.split("-").reversed().joinToString(separator = ".")
+            dateOfBirthFormatted.parseServerFormat().toDayFormat()
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/CwaUserCard.kt:46]: Birthdate here has special cases, birthdate can have the following cases:
- yyyy only ex: 1990
- yyyy-MM ex: 1990-10
- yyyy-MM-dd ex: 1990-10-01
So using standard formatter won't work as expected

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/CwaUserCard.kt:46]: @marcauberer  FYI ☝️ ?

Comment (marcauberer) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/CwaUserCard.kt]: Should be fixed now 👍🏼

**Comment by mtwalli** [comment]
Please rebase to 2.10.x ?  it is late for 2.9.0

**Comment by MikeMcC399** [comment]
Please check screenshots with birthdate where the header does not say YYYY-MM-DD, such as in PersonDetailsFragment_booster.png.

With locale set to English (United States) the date of birth is shown as 1981/02/20. This is not right. It should be 20/02/1981 (= Feb 20, 1981).

**Review by mtwalli — APPROVED** [review]
Lgtm, Thanks!

**Review by jurajkusnier — APPROVED** [review]
LGTM

**Comment by MikeMcC399** [comment]
See issue #4014 for regression in date of birth formatting.

**Pull request merged**